Misc/NEWS.d/next/Library/2023-03-10-13-21-16.gh-issue-102578.-gujoI.rst

```diff
@@ -0,0 +1,2 @@
+Speed up sanity check for ``__setattr__`` and ``__delattr__`` for frozen
+``dataclass``\es by using ``set``-based name lookup rather than ``tuple``\s.
```

Comment:
This is better, but still doesn't explain exactly what is being sped up here. My question when reading this is, "Well, why should I care about how fast `__setattr__` and `__delattr__` are on frozen dataclasses, given that `__setattr__` and `__delattr__` always raise exceptions on frozen dataclasses? The benchmark you've given on the issue also doesn't answer this question -- it seems like it consists of very contrived idioms that users would be unlikely to use in real-world code.

If the answer is "not much important is being sped up here, actually", then we need to ask whether it's worth making this change at all :)

Reply:
Sorry for the unclear description. I updated the news entry and the Pitch section in the linked issue.

> My question when reading this is, "Well, why should I care about how fast `__setattr__` and `__delattr__` are on frozen dataclasses, given that `__setattr__` and `__delattr__` always raise exceptions on frozen dataclasses?

My answer is no. It will not always raise exceptions. I added an example in the news entry and the Pitch section in the linked issue.

An example use case for inheriting frozen `dataclass`es.

```python
@dataclass(frozen=True)
class FrozenBase:
    x: int
    y: int
    ... # N_FIELDS

class Foo(FrozenBase):
    def __init__(self, x, y, somevalue, someothervalue):
        super().__init__(x, y)
        self.somevalue = somevalue            # takes O(N_FIELDS)
        self.someothervalue = someothervalue  # takes O(N_FIELDS) time again

foo = Foo(1, 2, 3, 4)
foo.extravalue = extravalue  # takes O(N_FIELDS) time again
```

Reply:
Thank you! The subclass case seems like a valid use case. I now agree that this seems like a good idea.